app/script/conversation/ConversationMapper.js

```diff
@@ -313,31 +331,34 @@ z.conversation.ConversationMapper = class ConversationMapper {
-      if (!mergedConversation.last_server_timestamp || wrong_server_timestamp) {
+      const wrongServerTimestamp = mergedConversation.last_server_timestamp < mergedConversation.last_event_timestamp;
+      if (!mergedConversation.last_server_timestamp || wrongServerTimestamp) {
         mergedConversation.last_server_timestamp = mergedConversation.last_event_timestamp;
       }
 
+      const isRemoteTimestampNewer = (localTimestamp, remoteTimestamp) => {
+        return localTimestamp !== undefined && remoteTimestamp > localTimestamp;
+      };
+
       // Some archived timestamp were not properly stored in the database.
       // To fix this we check if the remote one is newer and update our local timestamp.
-      const {archived_state: local_archived_state, archived_timestamp: local_archived_timestamp} = mergedConversation;
-      const remote_archived_timestamp = new Date(members.self.otr_archived_ref).getTime();
-      const is_remote_archived_timestamp_newer =
-        local_archived_timestamp !== undefined && remote_archived_timestamp > local_archived_timestamp;
+      const {archived_state: localArchivedState, archived_timestamp: localArchivedTimestamp} = mergedConversation;
+      const remoteArchivedTimestamp = new Date(members.self.otr_archived_ref).getTime();
+      const isRemoteArchivedTimestampNewer = isRemoteTimestampNewer(localArchivedTimestamp, remoteArchivedTimestamp);
 
-      if (is_remote_archived_timestamp_newer || local_archived_state === undefined) {
+      if (isRemoteArchivedTimestampNewer || localArchivedState === undefined) {
         mergedConversation.archived_state = members.self.otr_archived;
-        mergedConversation.archived_timestamp = remote_archived_timestamp;
+        mergedConversation.archived_timestamp = remoteArchivedTimestamp;
       }
 
-      const {muted_state: local_muted_state, muted_timestamp: local_muted_timestamp} = mergedConversation;
-      const remote_muted_timestamp = new Date(members.self.otr_muted_ref).getTime();
-      const is_remote_muted_timestamp_newer =
-        local_muted_timestamp !== undefined && remote_muted_timestamp > local_muted_timestamp;
+      const {muted_state: localMutedState, muted_timestamp: localMutedTimestamp} = mergedConversation;
+      const remoteMutedTimestamp = new Date(members.self.otr_muted_ref).getTime();
+      const isRemoteMutedTimestampNewer = isRemoteTimestampNewer(localMutedTimestamp, remoteMutedTimestamp);
 
-      if (is_remote_muted_timestamp_newer || local_muted_state === undefined) {
-        mergedConversation.muted_state = members.self.otr_muted;
-        mergedConversation.muted_timestamp = remote_muted_timestamp;
+      if (isRemoteMutedTimestampNewer || localMutedState === undefined) {
+        const hasMutedStatus = members.self.otr_muted_status !== null;
+        mergedConversation.muted_state = hasMutedStatus ? members.self.otr_muted_status : members.self.otr_muted;
```

Comment:
```js
mergedConversation.setNotificationsState(hasMutedStatus ? members.self.otr_muted_status : members.self.otr_muted);
```

Reply:
No. Merged conversation is not an entity. It is a plain object that we want to map into an entity later on.